yarn.config.cjs

```diff
@@ -0,0 +1,273 @@
+ * module resolution.
+ *
+ * @param {Workspace} workspace - The workspace to check.
+ * @returns {void}
+ */
+function expectExports(workspace) {
+  const { exports: manifestExports } = workspace.manifest;
+  Object.entries(manifestExports)
+    .filter(([, exportValue]) => typeof exportValue !== 'string')
+    .forEach(([exportName, exportObject]) => {
+      const keys = Object.keys(exportObject);
+      if (keys.includes('types') && keys[0] !== 'types') {
+        workspace.error(
+          `The "types" export must be the first export in the "exports" field for the export "${exportName}".`,
+        );
+      }
+    });
+}
+
+module.exports = defineConfig({
+  async constraints({ Yarn }) {
+    const workspace = Yarn.workspace();
+    const workspaceName = basename(__dirname);
+    const workspaceRepository = `${BASE_URL}${workspaceName}`;
+
+    // The package must have a name, version, description, and license.
+    expectWorkspaceField(workspace, 'name', `@metamask/${workspaceName}`);
+    expectWorkspaceField(workspace, 'version');
+    expectWorkspaceField(workspace, 'license');
+    expectWorkspaceDescription(workspace);
+
+    // The package must have a valid README.md file.
+    await expectReadme(workspace, workspaceName);
+
+    // The package must have a valid pull request template.
+    await expectPullRequestTemplate(workspace, workspaceName);
+
+    expectWorkspaceDependencies(workspace);
+
+    // The homepage of the package must match its name.
+    workspace.set('homepage', `${workspaceRepository}#readme`);
+
+    // The bugs URL of the package must point to the Issues page for the
+    // repository.
+    workspace.set('bugs.url', `${workspaceRepository}/issues`);
+
+    // The package must specify Git as the repository type, and match the URL of
+    // a repository within the MetaMask organization.
+    workspace.set('repository.type', 'git');
+    workspace.set('repository.url', `${workspaceRepository}.git`);
+
+    // The package must specify a minimum Node.js version of 16.
+    workspace.set('engines.node', '^18.18 || >=20');
+
+    // The package must specify a `types` entrypoint, and an `import`
+    // entrypoint.
+    workspace.set('types', './dist/types/index.d.ts');
+    workspace.set('exports["."].types', './dist/types/index.d.ts');
+
+    // The package must specify a `main` and `module` entrypoint, and a
+    // `require` and `import` entrypoint.
+    workspace.set('main', './dist/index.js');
+    workspace.set('exports["."].require', './dist/index.js');
+    workspace.set('module', './dist/index.mjs');
+    workspace.set('exports["."].import', './dist/index.mjs');
+
+    // The package must export a `package.json` file.
+    workspace.set('exports["./package.json"]', './package.json');
+
+    expectExports(workspace);
+
+    // The list of files included in the package must only include files
+    // generated during the build process.
+    workspace.set('files[0]', 'dist');
```

Comment:
It looks like this says that `dist` must be the first item in `files`. Does that matter? Should we say that `dist` must be present in `files` instead?

Reply:
Order of `files` doesn't matter. I've changed it to `workspace.set('files', ['dist'])` to make this clear. We could also check if the array contains `dist`, but I don't think that's possible with `workspace.set`.

Reply:
I think that's okay. Projects can customize this if they really need to, but I imagine that will be rare.